File: 1089 - Duplicate Zeros.cpp
#include <vector>
#include <queue>

using namespace std;

class Solution {
public:
    void duplicateZeros(vector<int>& arr) {
        queue<int> q;
        for (int i = 0; i < arr.size(); i++) {
            if (arr[i] == 0) {
                q.push(0);
                q.push(0);
            } else {
                q.push(arr[i]);
            }
            arr[i] = q.front();
            q.pop();
        }
    }
};

File: 1089 - Duplicate Zeros.java
import java.util.ArrayDeque;
import java.util.Queue;

class Solution {
    public void duplicateZeros(int[] arr) {
        Queue<Integer> q = new ArrayDeque<>();
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == 0) {
                q.add(0);
                q.add(0);
            } else {
                q.add(arr[i]);
            }
            arr[i] = q.poll();
        }
    }
}